### PR TITLE
mep12 on cursor_demo.py

### PR DESCRIPTION
**Pull request by ericmjl**
This is the last of them from the `pylab_examples` folder!

### CODE REVIEW

File: examples/pylab_examples/cursor_demo.py [hunk 1]
```diff
@@ -69,14 +68,14 @@ def mouse_move(self, event):
...
-s = sin(2*2*pi*t)
-fig, ax = subplots()
+t = np.arange(0.0, 1.0, 0.01)
+s = np.sin(2*2*np.pi*t)
+fig, ax = plt.subplots()
 
-cursor = Cursor(ax)
-#cursor = SnaptoCursor(ax, t, s)
-connect('motion_notify_event', cursor.mouse_move)
+# cursor = Cursor(ax)
```
Comment (thisch) [inline examples/pylab_examples/cursor_demo.py]: Don't use a space after a hash when commenting code in the examples. This convention is used in all examples, which I tried to follow in my pep8 PRs from last year.

Comment (ericmjl) [inline examples/pylab_examples/cursor_demo.py]: My bad, it's sublime text at work. Fixing on next push.

**Comment by jenshnielsen** [comment]
I would suggest adding a reference to mpldatacursor in the comments of this example somewhere https://github.com/joferkington/mpldatacursor

File: examples/pylab_examples/cursor_demo.py [hunk 2]
```diff
@@ -69,14 +68,14 @@ def mouse_move(self, event):
         print('x=%1.2f, y=%1.2f' % (x, y))
         draw()
```
Comment (WeatherGod) [inline examples/pylab_examples/cursor_demo.py]: This needs to be a `plt.draw()`

Comment (WeatherGod) [inline examples/pylab_examples/cursor_demo.py]: also, `searchsorted` above needs to be `np.searchsorted`

**Comment by ericmjl** [comment]
All done guys! This is the last example under the `pylab_examples` folder where pylab imports take place. :smile:

File: examples/pylab_examples/cursor_demo.py [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
 
 """
-
 This example shows how to use matplotlib to provide a data cursor.  It
 uses matplotlib to draw the cursor and may be a slow since this
 requires redrawing the figure with every mouse move.
 
 Faster cursoring is possible using native GUI drawing, as in
-wxcursor_demo.py
+wxcursor_demo.py.
```
Comment (jenshnielsen) [inline examples/pylab_examples/cursor_demo.py]: Unfortunately there is a trailing whitespace here now causing the pep8 test to fail

**Comment by ericmjl** [comment]
Okay! I hope there's no more pep8 failures!

**Comment by jenshnielsen** [comment]
Great, Thanks for all the work @ericmjl 

If you have time you could update the MEP document at https://github.com/matplotlib/matplotlib/blob/master/doc/devel/MEP/MEP12.rst with your progress. 

@matplotlib/developers should we consider renaming `pylab_examples` to `pyplot_examples` now when pylab is no longer used? In the long term the examples should be moved somewhere else but renaming the folder seems like something we can do for 1.5. but perhaps that just causes more confusion?

**Comment by OceanWolf** [comment]
I did wonder why we call this ``pylab_examples``, as I don't see much difference between these examples and the other ones.

Because of this, I would rather see these examples merged into other folders, for example this one could sit quite happily in ``event_handling``, ``ticks_and_spines``, or ``user_interfaces``.  Perhaps in the future we could find a way of putting the examples into multiple categories?

**Comment by jenshnielsen** [comment]
@OceanWolf Please read MEP12. That is what this is all about! 

But we cannot make that before 1.5 is released. So the question is only if the folder should be renamed to pyplot_examples for 1.5/2.0 (which is now more accurate)  and then then subsequently move the examples or if we should keep the folder name as pylab_examples and then move the files after 1.5/2.0

**Comment by OceanWolf** [comment]
I thought so, I just think that while slightly better, it still makes things sound more complicated...
How about renaming this directory ``other pyplot examples``?  Perhaps even leave out the ``pyplot`` part from that title?

**Comment by tacaswell** [comment]
I am a bit hesitant to move around the examples as it causes link-rot (because the url is derived from the folder name).  We probably should move them, but need a plan to provide forwarding links.

**Pull request merged**